### DISCUSSION TITLE
Animated.View component misbehaving when adding entering and leaving transitions

### DISCUSSION THREAD

**Discussion by AlexMachin1997** [software-mansion/react-native-reanimated#6756] [Q&A]
Hi everyone,

It's my first time posting here so if I'm missing anything let me know I'm happy to provide any additional details 😃 

I'm just in the process of going through an Airbnb clone using expo sdk v52 with the new architecture enabled and I noticed something quite odd with how the Animated.view component works.

So on one of my screens I've got a fixed footer which always sits at the bottom of my screen, this works fine until you add an entering or exiting. 

**How am using Animated.View component for my fixed footer:**
   ```
   <Animated.View
        style={{
          position: "absolute",
          height: 100,
          bottom: 0,
          left: 0,
          right: 0,
          backgroundColor: "#fff",
          paddingVertical: 10,
          paddingHorizontal: 20,
          borderTopColor: Colors.grey,
          borderTopWidth: StyleSheet.hairlineWidth,
        }}
        entering={SlideInDown.delay(200)}
      >
        <View
          style={{
            flexDirection: "row",
            justifyContent: "space-between",
            alignItems: "center",
          }}
        >
          <TouchableOpacity style={styles.footerText}>
            <Text style={styles.footerPrice}>€{listing.price}</Text>
            <Text>night</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={[defaultStyles.btn, { paddingRight: 20, paddingLeft: 20 }]}
          >
            <Text style={defaultStyles.btnText}>Reserve</Text>
          </TouchableOpacity>
        </View>
      </Animated.View>
```

**With entering/leaving animations:**

You can see it's not gone all the way down to the bottom of the screen, it's stuck part of the way up.

![Screenshot_20241124-165001](https://github.com/user-attachments/assets/fba0f5a0-d0f5-48da-950d-0cd34aacb606)

**Without entering/leaving animations:**

This is how it should look once the animation is complete.

![Screenshot_20241124-165023](https://github.com/user-attachments/assets/285a3469-d0bc-4adf-bce8-a80751c92379)

Has anyone else come across a similar issue whilst using an animated view with or without using "position: absolute" ? It seems to get stuck there until you cause a re-render then it sometimes fixed it self somehow.

Thanks,

Alex M

**Comment by MatiPl01** [comment]
Hey @AlexMachin1997!

Your problem looks familiar to me. Can you check which navigator does this app use? If it's a `@react-navigation/native-stack`, the issue is caused by the top-screen header measurement. The height of the header might not be available when the layout animation starts, hence, the layout animation can look weird and start/finish at the wrong position. In such a case, you can check if the issue still persists when you disable the header in the navigator.

You can disable header in this way:

```tsx
<Stack.Navigator
  ...
  screenOptions={{
    ...
    header: () => null, // add this
  }}>
  ...
</Stack.Navigator>
```

Let me know if the issue is related to the `@react-navigation/native-stack` or if it is a separate problem.

**Reply by dcorbin** [reply]
I have something that _may_ be somewhat releted.  I'm adding and removing several small Animated.Views with FadeIn and FadeOut.  Mostly it works fine.  But sometimes, say 1 in 50 times, the View ends in the wrong position.   Mind you, there is no animation of the position in any form.  It is an absolute position at a specific left/top.    One time I actually saw it moving up and down.  One time it appeared in the wrong space and later than it's "siblings".

Inspection the react-devtools shows nothing wrong with it that I can see, and when the item is selected in react-devtools, it highlights the position where the view is expected to be.

I am using Stack.Navigator, but there is no header on this view.

Saw it again, it to be honest it looks a little bit like a different KeyFrame Layout animation that I am using elsewhere in the app on a different view.   Are Animated.Views or animations cached in some way that might lead to accidentally being mis-applied?

**Reply by MatiPl01** [reply]
Thanks @dcorbin for your detailed description!

The issue you described is a separate problem. It is likely caused by the react's view recycling. When react reuses the same view, the layout animation may not be applied properly or it may even be applied on a view which doesn't have any layout animation assigned.

This behavior is a little bit tricky but we are already aware of the issue and are working on a solution.

**Reply by dcorbin** [reply]
Better news than I hoped. Is there any mitigation strategy until the problem is solved?  Can you provide a link to the issue so that I can track it?  Thanks.

**Reply by MatiPl01** [reply]
Sorry for the late response. There is no public issue where you can track the progress. There are some other layout animation problems that we are working on so this will be possibly fixed together with them. It may take some time so it's hard to estimate when fix will be ready.

**Reply by dcorbin** [reply]
No worries on the response delay.  I just assumed the holiday.  If you don't object, I will create an issue to cover it and post it here, because of course I want to be sure to know when it it fixed.

**Reply by MatiPl01** [reply]
Yeah, you can create the issue. I think we could have to discuss what is the best way to track progress so that it is visible to other people as well.

**Comment by dcorbin** [comment]
Issue created: https://github.com/software-mansion/react-native-reanimated/issues/6869
